tests/workbenches/notebook-controller/conftest.py

```diff
@@ -0,0 +1,282 @@
+        return
+
+    namespace = py_config["applications_namespace"]
+    config_map_name = "kueue-manager-config"
+
+    # Get the existing ConfigMap
+    try:
+        config_map = ConfigMap(
+            client=admin_client,
+            name=config_map_name,
+            namespace=namespace,
+            ensure_exists=True,
+        )
+    except Exception as e:
+        LOGGER.warning(f"Could not find kueue-manager-config ConfigMap: {e}")
+        LOGGER.info("This is expected for Red Hat Kueue operator scenario")
+        yield None
+        return
+
+    # Store original data and annotations for restoration
+    original_data = config_map.instance.data.copy() if config_map.instance.data else {}
+    original_annotations = (
+        config_map.instance.metadata.annotations.copy() if config_map.instance.metadata.annotations else {}
+    )
+
+    LOGGER.info("Storing original kueue-manager-config data for restoration")
+
+    # Get current config data
+    current_data = config_map.instance.data or {}
+    config_yaml = current_data.get("controller_manager_config.yaml", "{}")
+
+    # Parse the existing configuration
+    try:
+        config_dict = yaml.safe_load(config_yaml) or {}
+    except yaml.YAMLError:
+        config_dict = {}
+
+    # Ensure integrations section exists
+    if "integrations" not in config_dict:
+        config_dict["integrations"] = {}
+
+    if "frameworks" not in config_dict["integrations"]:
+        config_dict["integrations"]["frameworks"] = []
+
+    # Add pod and statefulset if not already present
+    frameworks = config_dict["integrations"]["frameworks"]
+    if "pod" not in frameworks:
+        frameworks.append("pod")
+    if "statefulset" not in frameworks:
+        frameworks.append("statefulset")
+
+    # Convert back to YAML
+    updated_config_yaml = yaml.dump(config_dict, default_flow_style=False)
+    updated_data = {**current_data, "controller_manager_config.yaml": updated_config_yaml}
+
+    # Apply the patch with both data and metadata annotations
+    patch = {"metadata": {"annotations": {Annotations.OpenDataHubIo.MANAGED: "false"}}, "data": updated_data}
+
+    def restart_kueue_deployment(reason: str):
+        """Helper function to restart the kueue-controller-manager deployment"""
+        if not should_restart_kueue_deployment(scenario):
+            LOGGER.info(f"Skipping kueue-controller-manager deployment restart for scenario: {scenario}")
+            return
+
+        LOGGER.info(f"Restarting kueue-controller-manager deployment - {reason}")
+
+        try:
+            kueue_deployment = Deployment(
+                client=admin_client,
+                name="kueue-controller-manager",
+                namespace=namespace,
+                ensure_exists=True,
+            )
+
+            # Get current replica count before restart
+            current_replicas = kueue_deployment.replicas
+            if current_replicas is None:
+                current_replicas = 1
```

Comment:
Is this possible?

Reply:
hehe :slightly_smiling_face: TBH, I added it here during my debugging as sometimes it happened that the deployment didn't spin up back for me - maybe some timing issues or anything... this was to be sure that we are always scaling back to 1